parser-typechecker/src/Unison/Typechecker/Extractor.hs

```diff
@@ -241,8 +240,7 @@ unknownSymbol = cause >>= \case
 unknownTerm :: Var v => ErrorExtractor v loc (loc, v, [C.Suggestion v loc], C.Type v loc)
 unknownTerm = cause >>= \case
   C.UnknownTerm loc v suggestions expectedType -> do
-    let k = Var.Inference Var.Ability
-        cleanup = Type.cleanup . Type.removePureEffects . Type.generalize' k
+    let cleanup = Type.cleanup . Type.removePureEffects
```

Comment:
@pchiusano It seemed strange to me that only variables of a specific kind were generalized. Nevertheless, I don't mean to undo some useful work of yours. Do you by any chance remember the motivation for that, or can you come up with an example that broke/got ugly after removing this generalization of inferred ability variables?

(This is why this is a draft PR for now.)